mirascope/core/anthropic/call_params.py

```diff
@@ -39,3 +40,19 @@ class AnthropicCallParams(BaseCallParams):
     top_k: NotRequired[int | None]
     top_p: NotRequired[float | None]
     timeout: NotRequired[float | Timeout | None]
+
+
+def get_anthropic_call_params_from_common(
```

Comment:
i know we can't attach methods to a typed dict, but I don't love the current structure of the implementation.

What if the user didn't have to do anything at all? Could we detect `CommonCallParams` under the hood and automatically convert? And then type hints / lint errors can take care of the rest?

What about something like this:

```python
from mirascope.core import anthropic

@anthropic.call(
    "claude-3-5-sonnet-latest",
    call_params={"temperature": 0.7},
)
def recommend_book(genre: str) -> str:
    return f"Recommend a {genre} book"
```

The above example will fail because Anthropic requires `max_tokens`. However, if we allow `call_params` to also take `CommonCallParams` as input, we could detect that this needs to be converted and convert it under the hood.

Then all of our conversion methods can live in each provider's respective `_utils` folder and the user never needs to worry about them.